internal/service/console.go

```diff
@@ -0,0 +1,40 @@
+package service
+
+import (
+	"context"
+
+	"github.com/flightctl/flightctl/internal/api/server"
+	"github.com/flightctl/flightctl/internal/flterrors"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/google/uuid"
+)
+
+func (h *ServiceHandler) RequestConsole(ctx context.Context, request server.RequestConsoleRequestObject) (server.RequestConsoleResponseObject, error) {
+	orgId := store.NullOrgId
+
+	// make sure the device exists
+	_, err := h.store.Device().Get(ctx, orgId, request.Name)
+	if err != nil {
+		switch err {
+		case flterrors.ErrResourceNotFound:
+			return server.RequestConsole404JSONResponse{}, nil
+		default:
+			return nil, err
+		}
+	}
+
+	sessionId := uuid.New().String()
+
+	annotations := map[string]string{"flightctl.io/console": sessionId}
```

Comment:
If there is a new request to connect to a console from a new client, and there is an existing console client , will the annotation be overwritten?

Reply:
Yes. More or less what we were discussing here: https://github.com/flightctl/flightctl/pull/313#discussion_r1671981031

In the final solution we shall account for timeouts, clearing out the annotations.  May be refuse to create a session when an ongoing session is running, and offer a --force method / query flag.